### PR TITLE
RE-1830 Ignore aborted uploads

### PR DESCRIPTION
**Pull request by hughsaunders**
Don't create an artifact upload failure issue if the build has been
aborted.

Issue: [RE-1830](https://rpc-openstack.atlassian.net/browse/RE-1830)

### CODE REVIEW

File: pipeline_steps/common.groovy [hunk 1]
```diff
@@ -1626,6 +1627,19 @@ Boolean isUserAbortedBuild() {
...
+  if (currentBuild.rawBuild.getAction(GhprbCancelBuildsOnUpdate.class)) {
+    prAborted = true
+  } else {
+    prAborted = false
+  }
+  return prAborted
+}
+
+Boolean isAbortedBuild(){
+  return isUserAbortedBuild || isPrUpdateAbortedBuild()
```
Comment (git-harry) [inline pipeline_steps/common.groovy]:
```suggestion
  return isUserAbortedBuild() || isPrUpdateAbortedBuild()
```

File: pipeline_steps/common.groovy [hunk 2]
```diff
@@ -409,10 +410,10 @@ def archive_artifacts(Map args = [:]){
       // the build result to be set to failure.
       // try-catch placed here because this function is used in multiple places.
       String jiraProjectKey = args.get("jiraProjectKey", "RE")
-      if (jiraProjectKey){
+      print "Error while archiving artifacts, swallowing this exception to prevent "\
```
Comment (git-harry) [inline pipeline_steps/common.groovy]: It might make sense to turn the if block on line 425 to and if/else block and add this to the else.

**Comment by hughsaunders** [comment]
recheck_whisperer

**Comment by hughsaunders** [comment]
whisperer failed on git timeout

**Comment by hughsaunders** [comment]
if this merges first: https://github.com/rcbops/rpc-gating/pull/1161/files then another instance of `isUserAbortedBuild()` will need to be modified.

**Pull request merged**